src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeController.cs

```diff
@@ -37,38 +37,56 @@ internal sealed class EventPipeController
-        private static readonly string[] ConfigFileLineDelimiters = new string[] { "\r\n", "\n" };
         private const char ConfigEntryDelimiter = '=';
 
+        private static char[] ProviderConfigDelimiter => Defaults.ProviderConfigDelimiter;
+        private static char[] ConfigComponentDelimiter => Defaults.ConfigComponentDelimiter;
+        private static string[] ConfigFileLineDelimiters => Defaults.ConfigFileLineDelimiters;
+
         // Config file keys.
-        private const string ConfigKey_Providers = "Providers";
-        private const string ConfigKey_CircularMB = "CircularMB";
-        private const string ConfigKey_OutputPath = "OutputPath";
-        private const string ConfigKey_ProcessID = "ProcessID";
-        private const string ConfigKey_MultiFileSec = "MultiFileSec";
+        private static string ConfigKey_Providers => Defaults.ConfigKey_Providers;
+        private static string ConfigKey_CircularMB => Defaults.ConfigKey_CircularMB;
+        private static string ConfigKey_OutputPath => Defaults.ConfigKey_OutputPath;
+        private static string ConfigKey_ProcessID => Defaults.ConfigKey_ProcessID;
+        private static string ConfigKey_MultiFileSec => Defaults.ConfigKey_MultiFileSec;
 
         // The default set of providers/keywords/levels.  Used if an alternative configuration is not specified.
-        private static readonly EventPipeProviderConfiguration[] DefaultProviderConfiguration = new EventPipeProviderConfiguration[]
+        private static EventPipeProviderConfiguration[] DefaultProviderConfiguration => Defaults.ProviderConfiguration;
+
+        // Inner static type to hold static initalization to avoid creating a static .cctor that's activated in the startup path
+        private static class Defaults
         {
-            new EventPipeProviderConfiguration("Microsoft-Windows-DotNETRuntime", 0x4c14fccbd, 5, null),
-            new EventPipeProviderConfiguration("Microsoft-Windows-DotNETRuntimePrivate", 0x4002000b, 5, null),
-            new EventPipeProviderConfiguration("Microsoft-DotNETCore-SampleProfiler", 0x0, 5, null),
-        };
+            internal static readonly EventPipeProviderConfiguration[] ProviderConfiguration = new EventPipeProviderConfiguration[]
+            {
+                new EventPipeProviderConfiguration("Microsoft-Windows-DotNETRuntime", 0x4c14fccbd, 5, null),
+                new EventPipeProviderConfiguration("Microsoft-Windows-DotNETRuntimePrivate", 0x4002000b, 5, null),
+                new EventPipeProviderConfiguration("Microsoft-DotNETCore-SampleProfiler", 0x0, 5, null),
+            };
+
+            internal static readonly char[] ProviderConfigDelimiter = new char[] { ',' };
+            internal static readonly char[] ConfigComponentDelimiter = new char[] { ':' };
+            internal static readonly string[] ConfigFileLineDelimiters = new string[] { "\r\n", "\n" };
+
+            // Config file keys.
```

Comment:
What's the point of moving the const strings behind extra layer of indirection? const strings are "free" - they do not cause any code to run at runtime.

Reply:
Moved back to `const`s